packages/alfa-rules/test/sia-r83/rule.spec.tsx

```diff
@@ -125,70 +130,74 @@ test(`evaluate() fails a text node that clips overflow by not wrapping text
 });
 
 test("evaluate() is inapplicable to a text node that is not visible", async (t) => {
-  const document = Document.of([
-    <div
-      style={{
-        overflow: "hidden",
-        whiteSpace: "nowrap",
-      }}
-      hidden
-    >
-      Hello world
-    </div>,
-  ]);
+  const document = Document.of(
+    [<div hidden>Hello world</div>],
+    [
+      h.sheet([
+        h.rule.style("div", {
+          overflow: "hidden",
+          whiteSpace: "nowrap",
+        }),
+      ]),
+    ]
+  );
 
   t.deepEqual(await evaluate(R83, { document }), [inapplicable(R83)]);
 });
 
 test(`evaluate() is inapplicable to a text node that is excluded from the
       accessibility tree using the \`aria-hidden\` attribute`, async (t) => {
-  const document = Document.of([
-    <div
-      style={{
-        overflow: "hidden",
-        whiteSpace: "nowrap",
-      }}
-      aria-hidden="true"
-    >
-      Hello world
-    </div>,
-  ]);
+  const document = Document.of(
+    [<div aria-hidden="true">Hello world</div>],
+    [
+      h.sheet([
+        h.rule.style("div", {
+          overflow: "hidden",
+          whiteSpace: "nowrap",
+        }),
+      ]),
+    ]
+  );
 
   t.deepEqual(await evaluate(R83, { document }), [inapplicable(R83)]);
 });
 
-test("evaluates() fails a clipped node with `normal` line-height", async (t) => {
-  const target = h.text("Hello World");
-
-  const document = Document.of([
-    <div
-      style={{
-        overflow: "hidden",
-        height: "1.5em",
-        fontSize: "16px",
-      }}
-    >
-      {target}
-    </div>,
-  ]);
+test(`evaluate() is inapplicable to a text node with a fixed absolute height set
+      via the style attribute`, async (t) => {
+  const document = Document.of(
+    [
+      <div
+        style={{
+          height: "20px",
+        }}
+      >
+        Hello world
+      </div>,
+    ],
+    [
+      h.sheet([
+        h.rule.style("div", {
+          overflow: "hidden",
+        }),
+      ]),
+    ]
+  );
 
-  t.deepEqual(await evaluate(R83, { document }), [
-    failed(R83, target, { 1: Outcomes.ClipsText }),
-  ]);
+  t.deepEqual(await evaluate(R83, { document }), [inapplicable(R83)]);
 });
 
-test(`evaluate() is inapplicable to a text node with a fixed width and hidden
-      overflow`, async (t) => {
-  const document = Document.of([
-    <div
-      style={{
-        overflow: "hidden",
-        width: "80ch",
-      }}
-    >
-      Hello world
-    </div>,
-  ]);
+test(`evaluate() is inapplicable to a text node with a fixed relative height`, async (t) => {
+  const document = Document.of(
+    [<div>Hello world</div>],
+    [
+      h.sheet([
+        h.rule.style("div", {
+          overflow: "hidden",
+          height: "1.2em",
+        }),
+      ]),
+    ]
+  );
 
```

Comment:
We should add an example for overflow clipped by an ancestor.

Reply:
The rule doesn't support those cases and can't reliably do so without a layout system. As mentioned, there has been several cases where fixed heights have been used in combination with `overflow-{x,y}: hidden` on an ancestor, sometimes near the root, and I've yet to see a case where this actually lead to clipped text. There has always some other thing going on in the layout that ensured that the text wouldn't clip. The rule therefore only deals with `overflow` and `height` properties specified for the same element.

Reply:
My bad 🙈 I thought this was also solving #731 (actually confused #622 with it when reading the names…)